packages/utils/src/abortableTimers.ts

```diff
@@ -2,32 +2,34 @@
  * setTimeout with AbortSignal support. Aborting will simply clear
  * the timeout silently.
  */
-export const setAbortableTimeout = createAbortableTimerFn(setTimeout, clearTimeout, true)
+export const setAbortableTimeout = (cb: () => void, ms: number, abortSignal: AbortSignal): void => {
```

Comment:
Before this PR the shared code between the functions was encapsulated to `createAbortableTimerFn`. Can we still use that, or is it necessary to duplicate the code?

Reply:
The problem with the previous implementation came from passing setTimeout and setInterval as parameters in runtime. This caused the virtual time used in simulations to break.

Reply:
Maybe could add a comment about this requirement and refactor the code to reduce some duplication